### PR TITLE
[DINOv2] Convert more checkpoints

### LINKED ISSUE #26167: Fail loading pretrained weights for  Dinov2ForImageClassification model
### System Info

both on:
transformers             4.32.0
transformers             4.34.0.dev0

### Who can help?

_No response_

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I've encountered a bug with the Dinov2ForImageClassification model from Hugging Face Transformers. As per the provided documentation [here](https://huggingface.co/docs/transformers/main/model_doc/dinov2#transformers.Dinov2ForImageClassification), I've followed the code example using the latest Transformers version. However, when running the code, I encounter an error indicating that the model is performing binary classification instead of the expected ImageNet 1000-way classification.

Here's my code:
```
from transformers import AutoImageProcessor, Dinov2ForImageClassification
import torch
from datasets import load_dataset

# Load a sample image dataset (in this case, 'huggingface/cats-image')
dataset = load_dataset('huggingface/cats-image')
image = dataset['test']['image'][0]

# Load the image processor and the Dinov2ForImageClassification model
image_processor = AutoImageProcessor.from_pretrained('facebook/dinov2-base')
model = Dinov2ForImageClassification.from_pretrained('facebook/dinov2-base')

# Prepare the input and obtain logits
inputs = image_processor(image, return_tensors='pt')
with torch.no_grad():
    logits = model(**inputs).logits

# The expected number of labels for ImageNet classification should be 1000
predicted_label = logits.argmax(-1).item()
```

Regardless of whether I specify num_labels=1000 during model initialization to correct the label dimensions, the following error persists:
```
Some weights of Dinov2ForImageClassification were not initialized from the model checkpoint at facebook/dinov2-base and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
```

The issue persists, and I'm unable to utilize the pretrained Dinov2ForImageClassification model for ImageNet 1000-way classification as intended.

### Expected behavior

loading without warning, having 1000-way long output vector, that is representing the correct classification labels of ImageNet.

see more here:
https://discuss.huggingface.co/t/dino2-for-classification-has-wrong-number-of-labels/55027

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR converts the DINOv2 checkpoints of image classification on ImageNet-1k ([source](https://github.com/facebookresearch/dinov2#pretrained-heads---image-classification)).

It also improves the doc test for `Dinov2ForImageClassification`.

Fixes #26167

### CODE REVIEW

File: src/transformers/models/dinov2/convert_dinov2_to_hf.py [hunk 1]
```diff
@@ -212,10 +242,14 @@ def convert_dinov2_checkpoint(model_name, pytorch_dump_folder_path, push_to_hub=
...
     if push_to_hub:
         model_name_to_hf_name = {
-            "dinov2_vits14": "dinov2-small",
-            "dinov2_vitb14": "dinov2-base",
-            "dinov2_vitl14": "dinov2-large",
-            "dinov2_vitg14": "dinov2-giant",
+            # "dinov2_vits14": "dinov2-small",
+            # "dinov2_vitb14": "dinov2-base",
+            # "dinov2_vitl14": "dinov2-large",
+            # "dinov2_vitg14": "dinov2-giant",
```
Comment (ArthurZucker) [inline src/transformers/models/dinov2/convert_dinov2_to_hf.py:248]:
```suggestion
```

**Review by ArthurZucker — APPROVED** [review]
I love the dinov2 family! Thanks for adding more models 😉

File: src/transformers/models/dinov2/convert_dinov2_to_hf.py (hunk 1 above)
Comment (NielsRogge) [reply src/transformers/models/dinov2/convert_dinov2_to_hf.py]: Sorry those were commented out but are actually needed

**Comment by ArthurZucker** [comment]
Thanks !
